### PR TITLE
copr: Add vectorized FromBase64

### LINKED ISSUE #6933: UCP: Migrate scalar function `FromBase64` from TiDB
## Description

Port the scalar function `FromBase64` from TiDB to coprocessor.

## Score 

* 50

## Mentor(s)

* @iosmanthus 

## Recommended Skills

* Rust programming 

## Learning Materials

Already implemented expressions ported from TiDB

- https://github.com/tikv/tikv/tree/master/components/tidb_query/src/rpn_expr) 
- https://github.com/tikv/tikv/tree/master/components/tidb_query/src/expr)

### PR DESCRIPTION
**Pull request by skyzh**
Signed-off-by: Alex Chi iskyzh@gmail.com

UCP #6933

Checklist #5751

### What problem does this PR solve?

Issue Number: close #6933 

Problem Summary: add vectorized FromBase64

### What is changed and how it works?

* Add vectorized FromBase64
* Move `strip_whitespace` to `tidb_query_shared_expr`

### Check List

Unit tested.

### CODE REVIEW

**Comment by skyzh** [comment]
This function is not in push-down test list. Should I make a PR to https://github.com/tikv/copr-test/blob/master/push-down-test/functions.txt , and run push-down test?

File: components/tidb_query_vec_expr/src/impl_string.rs [hunk 1]
```diff
@@ -492,6 +495,31 @@ pub fn to_base64(bs: &Option<Bytes>) -> Result<Option<Bytes>> {
...
+            let will_overflow = input_copy
+                .len()
+                .checked_mul(BASE64_INPUT_CHUNK_LENGTH)
+                .is_none();
+            // mysql will return "" when the input is incorrectly padded
+            let invalid_padding = input_copy.len() % BASE64_ENCODED_CHUNK_LENGTH != 0;
+            if will_overflow || invalid_padding {
+                Ok(Some(Vec::new()))
+            } else {
+                match base64::decode_config(&input_copy, base64::STANDARD) {
```
Comment (TennyZhuang) [inline components/tidb_query_vec_expr/src/impl_string.rs:513]: prefer use https://doc.rust-lang.org/std/result/enum.Result.html#method.ok

Ok(....ok())

**Comment by TennyZhuang** [comment]
rest LGTM

**Comment by skyzh** [comment]
It seems that integration test failed. Is it acceptable?

**Comment by TennyZhuang** [comment]
@sre-bot /test

**Comment by skyzh** [comment]
Integration common test reports
```
[2020-05-11T01:35:15.752Z] time="2020-05-11T09:35:15+08:00" level=fatal msg="run test [gcol_blocked_sql_funcs] err: sql:create table t1 (a varchar(32), b bool generated always as (match a against ('sample text')) virtual);: run \"create table t1 (a varchar(32), b bool generated always as (match a against ('sample text')) virtual);\" around line 80 err, we need(299):\ncreate table t1 (a varchar(32), b bool generated always as (match a against ('sample text')) virtual);\n[parser:1064]You have an error in your SQL syntax; check the manual that corresponds to your TiDB version for the right syntax to use line 1 column 65 near \"match a against ('sample text')) virtua\nbut got(299):\ncreate table t1 (a varchar(32), b bool generated always as (match a against ('sample text')) virtual);\n[parser:1064]You have an error in your SQL syntax; check the manual that corresponds to your TiDB version for the right syntax to use line 1 column 67 near \"a against ('sample text')) virtual);\" \n\n"
```
in integration MySQL test

**Comment by TennyZhuang** [comment]
It seems there are some bugs in the main branch.

**Comment by skyzh** [comment]
Thanks for reviewing! Maybe I can trigger CI build later today to see if it works.

**Pull request merged**